File: HillHigh1980.Core/ApplicationService/Service/RosterService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using HillHigh1980.Core.DomainService;
using HillHigh1980.Core.Entity;
using HillHigh1980.Core.Entity.Jut.Locations;
using HillHigh1980.Core.Entity.Jut.Rosters;

namespace HillHigh1980.Core.ApplicationService.Service
{
    public class RosterService : IRosterService
    {
        private readonly IRosterRepository _repository;

        public RosterService(IRosterRepository repository)
        {
            _repository = repository;
        }
        public async Task<RosterJut> FindRosterByIdAsync(int rosterId)
        {
            Roster roster = await _repository.FindById(rosterId);
            return roster.Jut();
        }

        public async Task<List<RosterJut>> GetAllRostersAsync()
        {
            IEnumerable<Roster> rosters = await _repository.ReadAll();

            return rosters.Jut().ToList();
        }

        public async Task<List<RosterJut>> FindRostersByName(Filter filter)
        {
            IEnumerable<Roster> rosters = await _repository.FindRostersByName(filter);
            return rosters.Jut().ToList();
        }

        public async Task<int> CreateRosterLocation(List<LocationJut> locations)
        {
            return await _repository.CreateLocation(locations.JutToLocation());
        }

        public async Task<List<LocationJut>> GetRosterLocations(int rosterId)
        {
            return (await _repository.GetLocations()).Jut().ToList();
        }

        public async Task<LocationJut> UpdateRosterLocation(LocationJut location)
        {
            return (await _repository.UpdateLocation(location.JutToLocation())).Jut();
        }

        public async Task RemoveRosterLocation(LocationJut location)
        {
            await _repository.DeleteLocation(location.JutToLocation());
        }
    }
}

File: HillHigh1980.UI/wwwroot/Script/Core/ApplicationService/Service/RosterService.ts
module GScope {
    export module ApplicationService {
        export module Service {
            export class RosterService implements IRosterService {
                private readonly _repository: RepositoryService.IRosterRepository;

                constructor(repository: RepositoryService.IRosterRepository) {
                    this._repository = repository;
                }

                FindRosterById(rosterId: number): Promise<Entity.Roster> {
                    return this._repository.FindById(rosterId);
                }
                FindRostersByName(filter: Entity.Filter): Promise<Entity.Roster[]> {
                    return this._repository.FindByName(filter);
                }
                GetAllRosters(): Promise<Entity.Roster[]> {
                    return this._repository.ReadAll();
                }

                CreateRosterLocations(locations: [Entity.Location]): Promise<any> {
                    return this._repository.CreateLocations(locations);
                }
                UpdateRosterLocation(location: Entity.Location): Promise<any> {
                    return this._repository.UpdateLocation(location);
                }
                DeleteRosterLocation(location: Entity.Location): Promise<any> {
                    return this._repository.DeleteLocation(location);
                }
            }
        }
    }
}